build/pluto/prometheus/exporters/blackbox.nix

```diff
@@ -37,30 +12,104 @@ in
             tcp.tls = true;
             http.headers.User-Agent = "blackbox-exporter";
           };
+
+          # From https://github.com/prometheus/blackbox_exporter/blob/53e78c2b3535ecedfd072327885eeba2e9e51ea2/example.yml#L120-L133
+          modules.smtp_starttls = {
+            prober = "tcp";
+            timeout = "5s";
+            tcp = {
+              query_response = [
+                { expect = "^220 ([^ ]+) ESMTP (.+)$"; }
+                { send = "EHLO prober\r"; }
+                { expect = "^250-STARTTLS"; }
+                { send = "STARTTLS\r"; }
+                { expect = "^220"; }
+                { starttls = true; }
+                { send = "EHLO prober\r"; }
+                { expect = "^250-AUTH"; }
+                { send = "QUIT\r"; }
+              ];
+            };
+          };
         }
       );
     };
 
     scrapeConfigs = [
-      (mkProbe "https_success" [
-        "https://cache.nixos.org"
-        "https://channels.nixos.org"
-        "https://common-styles.nixos.org"
-        "https://discourse.nixos.org"
-        "https://hydra.nixos.org"
-        "https://mobile.nixos.org"
-        "https://monitoring.nixos.org"
-        "https://nixos.org"
-        "https://planet.nixos.org"
-        "https://releases.nixos.org"
-        "https://status.nixos.org"
-        "https://survey.nixos.org"
-        "https://tarballs.nixos.org"
-        "https://weekly.nixos.org"
-        "https://wiki.nixos.org"
-        "https://www.nixos.org"
-        "https://tracker.security.nixos.org"
-      ])
+      {
+        job_name = "blackbox-https_success";
```

Comment:
I opted to remove the existing `mkProbe` function, because I would have had to complicate it to handle `smtp_starttls` *and* `https_success`.

Reply:
Yeah, we're not going to randomly make new mailservers appear, so I think `static_configs` is probably fine.

Reply:
Sure, I can make that change when I have some time. Just to make sure I understand your reasoning, though: is the reason to keep things simple and keep the old `mkProbe` function around?

Reply:
The relabeling config is what makes the configuration of the blackbox exporter scraping rule complicated, so I wouldn't want to repeat that for every probe. If you want to stick with dnssd create another function to keep the remaining config leaner.

Reply:
I would prefer to use the MX record. I've renamed the old `mkProbe` to `mkHttpsProbe`, and added a new `mkSmtpStartTlsProbe`. Please take another look!

Reply:
mkStaticProbe, mkDNSSDProbe

Reply:
Updated, PTAL.